src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -146,68 +146,63 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
-        n_streams = 1;
-        if ((proc_type_table.size() == 1) && (input_threads == 0) && (model_prefer_threads > 0)) {
-            stream_info[NUMBER_OF_STREAMS] = n_streams;
-            if ((model_prefer_threads == proc_type_table[0][MAIN_CORE_PROC]) &&
-                (proc_type_table[0][MAIN_CORE_PROC] > 0)) {
-                stream_info[PROC_TYPE] = MAIN_CORE_PROC;
-                n_threads_per_stream = proc_type_table[0][MAIN_CORE_PROC] + proc_type_table[0][HYPER_THREADING_PROC];
-                stream_info[THREADS_PER_STREAM] = n_threads_per_stream;
-                update_ids_method(proc_type_table[0]);
-            } else if (proc_type_table[0][MAIN_CORE_PROC] == 0) {
-                stream_info[PROC_TYPE] = EFFICIENT_CORE_PROC;
-                n_threads_per_stream = proc_type_table[0][EFFICIENT_CORE_PROC];
-                stream_info[THREADS_PER_STREAM] = n_threads_per_stream;
-                update_ids_method(proc_type_table[0]);
+        if (input_threads > 0) {
+            n_streams = 1;
+            n_threads_per_stream = std::min(input_threads, proc_type_table[0][ALL_PROC]);
+            if (proc_type_table.size() == 1) {
+                if ((n_threads_per_stream > proc_type_table[0][MAIN_CORE_PROC]) &&
+                    (proc_type_table[0][MAIN_CORE_PROC] > 0)) {
+                    stream_info[PROC_TYPE] = ALL_PROC;
+                }
+            }
+        } else if (((input_streams_changed == false) &&
+                    (latencyThreadingMode == Config::LatencyThreadingMode::PER_PLATFORM)) ||
+                   (proc_type_table.size() == 1) || ((input_streams_changed == true) && (input_streams == 1))) {
+            n_streams = 1;
+            n_threads_per_stream = std::min(input_threads, proc_type_table[0][ALL_PROC]);
+            if ((proc_type_table.size() == 1) && (model_prefer_threads > 0)) {
+                stream_info[NUMBER_OF_STREAMS] = n_streams;
+                if ((model_prefer_threads == proc_type_table[0][MAIN_CORE_PROC]) &&
+                    (proc_type_table[0][MAIN_CORE_PROC] > 0)) {
+                    stream_info[PROC_TYPE] = MAIN_CORE_PROC;
+                    n_threads_per_stream =
+                        proc_type_table[0][MAIN_CORE_PROC] + proc_type_table[0][HYPER_THREADING_PROC];
+                    stream_info[THREADS_PER_STREAM] = n_threads_per_stream;
+                    update_ids_method(proc_type_table[0]);
+                } else if (proc_type_table[0][MAIN_CORE_PROC] == 0) {
+                    stream_info[PROC_TYPE] = EFFICIENT_CORE_PROC;
+                    n_threads_per_stream = proc_type_table[0][EFFICIENT_CORE_PROC];
+                    stream_info[THREADS_PER_STREAM] = n_threads_per_stream;
+                    update_ids_method(proc_type_table[0]);
+                } else {
+                    stream_info[PROC_TYPE] = ALL_PROC;
+                    n_threads_per_stream = proc_type_table[0][ALL_PROC];
```

Comment:
If model_prefer_threads == proc_type_table[0][MAIN_CORE_PROC] + proc_type_table[0][EFFICIENT_CORE_PROC], all cores would be used?

Reply:
Yes. latency mode support using hyper threads when user set `enable_hyper_threading`.

`proc_type_table[0][HYPER_THREADING_PROC]` is updated by `enable_hyper_threading` in `apply_hyper_threading()`